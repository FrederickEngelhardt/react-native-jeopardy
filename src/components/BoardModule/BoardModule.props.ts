import categoryColumnProps from "../CategoryColumn/CategoryColumn.props";

const column2 = {
  ...categoryColumnProps,
  headlineCard: { title: "Planets" },
  updateUserScore: () => {}
}

const boardModuleProps = {
  categoryColumns: [{...categoryColumnProps, updateScore: () => {}}, column2]
};

export default boardModuleProps;
