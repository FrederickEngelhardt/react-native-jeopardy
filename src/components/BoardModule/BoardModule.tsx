import React from "react";
import styled, { withTheme } from "styled-components";
import CategoryColumn, { Props as CategoryColumnProps } from "../CategoryColumn/CategoryColumn";
import {Theme} from "../RootThemeProvider";

const CategoryScrollView = styled.ScrollView`
`

interface Props {
  categoryColumns: [CategoryColumnProps];
  theme: Theme;
}

interface State {
  activeCardId: number | null;
}

class BoardModule extends React.PureComponent<Props, State> {
  state = {
    activeCardId: null
  }
  render() {
    const { categoryColumns, theme } = this.props;

    const generateSnapOffsets = categoryColumns.map((ele, index: number) => (index * theme.deviceWidth))
    return (
      <CategoryScrollView
        horizontal
        snapToOffsets={generateSnapOffsets}
      >
        {categoryColumns.map((column) => (
          <CategoryColumn key={column.headlineCard.title} {...column} />
        ))}
      </CategoryScrollView>
    );
  }
}

export default withTheme(BoardModule);
