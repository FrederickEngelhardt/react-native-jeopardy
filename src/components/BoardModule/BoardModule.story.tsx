import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import BoardModule from "./BoardModule";
import boardModuleProps from "./BoardModule.props";
import { withKnobs } from "@storybook/addon-knobs";

storiesOf("BoardModule", module)
  .addDecorator(withKnobs)
  .add("default", () => <BoardModule {...boardModuleProps} />);
