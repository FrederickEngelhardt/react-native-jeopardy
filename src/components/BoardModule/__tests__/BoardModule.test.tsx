import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

import BoardModule from "../BoardModule";
import boardModuleProps from "../BoardModule.props";

const theme = {
  darkMode: true,
  deviceHeight: 800,
  deviceWidth: 320,
};

describe("test", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme} >
      <BoardModule {...boardModuleProps}/>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
