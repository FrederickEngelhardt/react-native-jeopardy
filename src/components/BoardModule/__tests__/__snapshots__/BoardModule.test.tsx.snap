// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardModule /> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    horizontal={true}
    snapToOffsets={
      Array [
        0,
        320,
      ]
    }
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 320,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,.8)",
              "marginBottom": 12,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffe81f",
                "fontSize": 48,
                "fontWeight": "900",
                "textAlign": "center",
              },
            ]
          }
        >
          Apprentices
        </Text>
      </View>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            100
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            200
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            300
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            400
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            500
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      style={
        Array [
          Object {
            "width": 320,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,.8)",
              "marginBottom": 12,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffe81f",
                "fontSize": 48,
                "fontWeight": "900",
                "textAlign": "center",
              },
            ]
          }
        >
          Planets
        </Text>
      </View>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            100
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            200
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            300
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            400
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Jeopardy Topic Card"
        accessible={true}
        activeOpacity={0.5}
        disabled={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "opacity": 1,
              "width": "80%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.8)",
                "borderRadius": 5,
                "marginBottom": 12,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffe81f",
                  "fontSize": 38,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
              ]
            }
          >
            500
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </ScrollView>
</View>
`;
